### PR TITLE
enabled web-vitals-extension loggin option

### PR DESCRIPTION
**Pull request by gilbertococchi**
Progress on #77

Added Console Log Options to enable all CWV Metrics logging via console.

### CODE REVIEW

**Comment by gilbertococchi** [comment]
Here's an example of the Console Logging output.

All the LCP and CLS intermediate values will be logged, also including the latest "entries" object to allow an easy identification of the DOM elements causing the metric timing.

**Comment by cheneytsai** [comment]
This is a great feature! Reduces the anxiety of having to click the extension at the right time without doing anything else that might interfere with what you wanted to measure!

File: src/browser_action/vitals.js
Comment (rviscomi) [inline src/browser_action/vitals.js:222]: Should each of these metrics' logs be moved to `broadcastMetricsUpdates`? That will also debounce the CLS logging to prevent console spam.

One other suggestion is to prefix the log message with something identifiable for filtering, eg `"[Web Vitals] "`

**Review by rviscomi — CHANGES_REQUESTED** [review]
Looks great, thanks @gilbertococchi! One comment.

File: src/browser_action/vitals.js
Comment (gilbertococchi) [reply src/browser_action/vitals.js]: Hi Rick, thanks for your feedback.

Yes I agree with you, at first I've added the logging within `broadcastMetricsUpdates` but then I've moved to the getMetricName method because I wanted to create one line per Metric logging.

I've moved the logging to that method, I thought about keeping the metric value rounded toFixed(2) in the first logging value and send as last log entry the whole metric object.

Maybe we can support different logging levels, such as debug (metric name + value) and verbose (metric name, value, metric object with all the entries).
WDYT?

Below the example screenshot
Screenshot 2020-11-17 at 19 58 09

Comment (rviscomi) [reply src/browser_action/vitals.js]: > Maybe we can support different logging levels, such as debug (metric name + value) and verbose (metric name, value, metric object with all the entries).
> WDYT?

I think the way you have it here is a good first step for this PR and we can keep #77 open to revisit it with enhancements like the one you suggested.

**Review by rviscomi — APPROVED** [review]
LGTM!

I'll defer to @addyosmani to give this one last look before merging.

**Review by addyosmani — APPROVED** [review]
LGTM. Thank you! 

We'll include this in our next release.

**Pull request merged**
